audio/src/audio.py

```diff
@@ -1,73 +1,161 @@
-import random
-import string
+import threading
 
 import os
+import time
+
 os.environ['PYGAME_HIDE_SUPPORT_PROMPT'] = "hide"
 import pygame as pg
 
 class AudioMixer:
 
-    def __init__(self, nof_max_sounds=20):
+    def __init__(self, event_callback, nof_max_sounds=20):
         pg.mixer.init()
         pg.init()
         pg.mixer.set_num_channels(nof_max_sounds)
+        self._event_callback = event_callback
         self._sounds = {}
+        self._sound_infos = {}
+        self._end_events = {}
+        self._current_end_event = 1000
+
+        self._event_handle_thread = threading.Thread(target=self._event_handler)
```

Comment:
I'm always nervous when threads are involved. If I understand the flow correctly, CoreConnection._ws is accessed both from the main thread and this thread. Is this safe?

Reply:
It mainly depends on whether the implementation of the websocket-client lib is thread safe. Multithread support is enabled by default and they have an info page about weird quirks found with threading (see [here](https://websocket-client.readthedocs.io/en/latest/threading.html)) and as far as I can tell it should be fine in this context. Good point though!